cmd/baseosmgr/baseosmgr.go

```diff
@@ -390,16 +390,19 @@ func handleCertObjModify(ctx *baseOsMgrContext, key string, config *types.CertOb
 	uuidStr := config.Key()
 	log.Infof("handleCertObjModify for %s\n", uuidStr)
 
-	if config.UUIDandVersion.Version == status.UUIDandVersion.Version {
-		log.Infof("Same version %v for %s\n",
-			config.UUIDandVersion.Version, key)
-		return
-	}
+	if config.UUIDandVersion.Version != status.UUIDandVersion.Version {
+		log.Infof("handleCertObjModify(%s), New config version %v\n", key,
+			config.UUIDandVersion.Version)
+		status.UUIDandVersion = config.UUIDandVersion
+		publishCertObjStatus(ctx, status)
 
-	status.UUIDandVersion = config.UUIDandVersion
-	publishCertObjStatus(ctx, status)
+	}
 
-	certObjHandleStatusUpdate(ctx, config, status)
+	// on storage config change, purge and recreate
+	if certObjCheckConfigModify(ctx, key, config, status) {
+		removeCertObjConfig(ctx, key)
+		handleCertObjCreate(ctx, key, config)
+	}
```

Comment:
Do we need an else branch which calls certObjHandleStatusUpdate() here?

Reply:
Its a config change, and mostly other than the cert objects, nothing else is there. We can add a else calling certObjHandleStatusUpdate(), but will be repeatative, harmless code path.

Reply:
OK